packages/block-library/src/latest-posts/index.php

```diff
@@ -5,6 +5,26 @@
  * @package WordPress
  */
 
+/**
+ * The excerpt length set by the Latest Posts core block
+ * set at render time and used by the block itself.
+ *
+ * @var int
+ */
+$block_core_latest_posts_excerpt = 0;
```

Comment:
Okay, I see now how the global variable is being used to account for the block attribute.

But:

- I don't know that we need to declare this variable here at all. If there's a default, it should be provided through the attributes schema.
- With PHP support now bumped, we might be able to implement this without resorting to a global variable.

I'm not sure if it works to add filters using a function assigned to a variable, but if it does, it may be possible to implement inline in `render_block_core_latest_posts ` as:

```php
$filter_excerpt_length = function( $length ) use ( $attributes ) {
	if ( ! empty( $attributes['excerptLength'] ) ) {
		return $attributes['excerptLength'];
	}

	return $length;
};

add_filter( 'excerpt_length', $filter_excerpt_length );

// ...

remove_filter( 'excerpt_length', $filter_excerpt_length );
```

Reply:
Related: https://make.wordpress.org/core/2019/03/26/coding-standards-updates-for-php-5-6/

>Anonymous functions must not be passed as filter or action callbacks in WordPress Core, as they cannot be removed by remove_action() / remove_filter() (see #46635 for a proposal to address this). Outside of Core, developers may pass anonymous functions as filter or action callbacks at their own discretion.

This specific usage is a bit unique though, in that the filter is only temporarily added for the purpose of controlling the behavior of excerpt length, then immediately removed. It's not something which we would expect plugins to be able to have access to referencing for removal anyways (IIRC, manipulating peer filters in the context of a filter callback will not have an effect within the same callback execution).